src/map/battlefield_handler.cpp

```diff
@@ -223,8 +225,10 @@ uint8 CBattlefieldHandler::LoadBattlefield(CCharEntity* PChar, const Battlefield
 
     if (!PChar->StatusEffectContainer->GetStatusEffect(EFFECT_BATTLEFIELD))
     {
+        // The first bit of the subpower is reserved for if the player has entered the battlefield or not
+        uint16 subpower = PBattlefield->GetArea() << 1;
```

Comment:
I seriously dislike setting this precedent, or using this pattern at all especially for this. Bitpacking/shifting to handle things in an effect field don't sit well with me.

I'd rather do something far messier tbh. 3 (4?) rewrites ago, players would get "registered" in the core to a battlefield copy on entry for its duration. I forget the exact code involved in that version of things.

Reply:
I'm genuinely curious: How terribad would it be to add new things to the battlefield object to toggle per copy+person?

Reply:
What we need this for is when the battlefield doesn't exist anymore and we need to kick the player out. Some alternatives include:
- Scrap the area since we aren't actively using it and allow the subpower to simply be entered or not (1 or 0).
- Use a charvar that we set when they enter and clear when they leave. This honestly wouldn't be too bad since it'll be cleared out when they log back in.

Reply:
Instead of storing this on the player in the effect, I wonder if it might be better to track this in the battlefield object itself, since _I think_ we already know they're registered?

Reply:
The other option is that we could potentially use status effect Tier as well

Reply:
> What we need this for is when the battlefield doesn't exist anymore and we need to kick the player out. Some alternatives include:
> 
> * Scrap the area since we aren't actively using it and allow the subpower to simply be entered or not (1 or 0).
> * Use a charvar that we set when they enter and clear when they leave. This honestly wouldn't be too bad since it'll be cleared out when they log back in.

one or more of the previous incarnations of bcnm systems used to do the charvar. to me it'd be better than bitpacking this into the effect. as for the battlefield not existing anymore, couldn't we use that in itself as a stand in - use the value in the object when the object does exist, use the fact it doesn't exist when it doesn't to mean "false" as well?

using tier as an in/out  (0 or 1) also is more appealing than using subpower here, if we go a 1 or 0 route

Reply:
The reason we cant store this on the battlefield object itself is because the character is logging out for an unknown amount of time. We wouldn't want to keep the battlefield instance around and even if we tried doing that there is always the scenario that the server goes down.

Using the tier looks easy enough and is a great idea! That's what I'm going to roll with.

Reply:
Tier is fine, as long as we ain't bitpacking I'm content

Reply:
> 
> Yes but if the battlefield ceased to exist, we wouldn't need to know if a player can enter it anymore - we can treat that as the same as "no you can't" if we check for that condition (nil/null) see what i am saying?

This is for the scenario that the player has logged out/disconnected while **inside** the battlefield. In this case we need to kick the player out. However, if they didn't enter but have the battlefield effect then we don't want to kick them out.

Reply:
> This is for the scenario that the player has logged out/disconnected while inside the battlefield. In this case we need to kick the player out. 

this is still "battlefield ceased to exist, you don't belong here /kick"

> However, if they didn't enter but have the battlefield effect then we don't want to kick them out.

????

how exactly do you have the effect and be inside a battlefield that ceased to exist

Reply:
Log out inside a battlefield and when you log back in you'll still have the effect